integration-cli/docker_cli_links_test.go

```diff
@@ -221,31 +182,32 @@ func (s *DockerSuite) TestLinkShortDefinition(c *check.C) {
 	out, _ := dockerCmd(c, "run", "-d", "--name", "shortlinkdef", "busybox", "top")
 
 	cid := strings.TrimSpace(out)
-	c.Assert(waitRun(cid), check.IsNil)
+	c.Assert(waitRun(cid), checker.IsNil)
 
 	out, _ = dockerCmd(c, "run", "-d", "--name", "link2", "--link", "shortlinkdef", "busybox", "top")
 
 	cid2 := strings.TrimSpace(out)
-	c.Assert(waitRun(cid2), check.IsNil)
+	c.Assert(waitRun(cid2), checker.IsNil)
 
 	links, err := inspectFieldJSON(cid2, "HostConfig.Links")
-	c.Assert(err, check.IsNil)
-	c.Assert(links, check.Equals, "[\"/shortlinkdef:/link2/shortlinkdef\"]")
+	c.Assert(err, checker.IsNil)
+	c.Assert(links, checker.Equals, "[\"/shortlinkdef:/link2/shortlinkdef\"]")
 }
 
 func (s *DockerSuite) TestLinksNetworkHostContainer(c *check.C) {
 	testRequires(c, DaemonIsLinux, NotUserNamespace)
 	dockerCmd(c, "run", "-d", "--net", "host", "--name", "host_container", "busybox", "top")
 	out, _, err := dockerCmdWithError("run", "--name", "should_fail", "--link", "host_container:tester", "busybox", "true")
-	if err == nil || !strings.Contains(out, "--net=host can't be used with links. This would result in undefined behavior") {
-		c.Fatalf("Running container linking to a container with --net host should have failed: %s", out)
-	}
+
+	// Running container linking to a container with --net host should have failed
+	c.Assert(err, checker.NotNil, check.Commentf("out: %s", out))
+	// Running container linking to a container with --net host should have failed
+	c.Assert(out, checker.Contains, "--net=host can't be used with links. This would result in undefined behavior")
 }
 
 func (s *DockerSuite) TestLinksEtcHostsRegularFile(c *check.C) {
 	testRequires(c, DaemonIsLinux, NotUserNamespace)
 	out, _ := dockerCmd(c, "run", "--net=host", "busybox", "ls", "-la", "/etc/hosts")
-	if !strings.HasPrefix(out, "-") {
-		c.Errorf("/etc/hosts should be a regular file")
-	}
+	// /etc/hosts should be a regular file
+	c.Assert(strings.HasPrefix(out, "-"), checker.Equals, true)
```

Comment:
`check.Matches` instead?
`c.Assert(out, check.Matches, '^-')`

Note I have not tested.
If Matches doesn't work, then need to make sure to put `out` in a Commentf.